rasa/server.py

```diff
@@ -149,6 +150,26 @@ def decorated(*args, **kwargs):
     return decorator
 
 
+def ensure_conversation_exists(app: Sanic) -> Callable[..., HTTPResponse]:
+    """Wraps a request handler ensuring the conversation exists.
+    """
+
+    def decorator(f: Callable[..., HTTPResponse]) -> HTTPResponse:
+        @wraps(f)
+        def decorated(*args: Any, **kwargs: Any) -> HTTPResponse:
+            conversation_id = kwargs["conversation_id"]
+            if app.agent.tracker_store.exists(conversation_id):
```

Comment:
You can also extract the agent from the request (which is your first arg) using

```
agent = request.app.agent
```

This allows you to skip passing in `app` needlessly